File: cpp/sort_babel.cpp
/*
 * sort_wybor.cpp
 */


#include <iostream>
#include <time.h>

using namespace std;

void wypelnij(int t[], int n, int max)
{   
    srand(time(NULL)); // inicjacja generatora liczb pseudolosowych
    for (int i=0; i<n; i++)
    {   
        t[i] = 1 + rand() % max; // losowanie liczb calkowitych < 0; maks>
        
        
    }
}

void drukuj(int t[], int n)
{   
    for (int i=0; i<n; i++)
    {   
       cout << t[i] << " ";
    }
    cout << endl;
}

void zamien(int &a, int &b) //zainicjowana zmienna zapamietujaca dana wartosc
{
    int tmp = a;
    a = b;
    b = tmp;
}


void sort_babel(int tab[], int n)
{
         //buuble sort
    cout<<"**************Sortowanie babelkowe**************" <<endl;
    for(int i = 0; i < n; i++)
    {
        for(int j = 0; j < n - 1 - i; j++)
        {
            if(tab[j] > tab[j+1])
                zamien(tab[j], tab[j+1]);
            
        }
        
    }
    
       
}

int main(int argc, char **argv)
{
	const int ile = 10;
    int tab[ile];
    wypelnij(tab, ile, 20);
    drukuj(tab, ile);
    sort_babel(tab, ile);
    drukuj(tab, ile);
	return 0;
}

File: python/sort_babel.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
#  sort_babel.py 
#  

from random import randint


def wypelnij(lista, ile, maks):
    for i in range(ile):
        lista.append(randint(0, maks))
    return lista


def sort_babel(lista):
    print("###########-SORTOWANIE BABELKOWE-###########")
    for i in range(len(lista) - 1, 0 ,-1):
        for j in range(i):
            if lista[j] > lista[j + 1]:
                lista[j], lista[j + 1] = lista[j + 1], lista[j]

    return lista
    

def main(args):
    lista = []
    print(wypelnij(lista, 10, 20))
    print(sort_babel(lista))
    return 0

if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))
